cmake/ProcessOptions.cmake

```
@@ -457,7 +457,7 @@ endfunction()
 function( NEST_PROCESS_WITH_MPI )
   # Find MPI
   set( HAVE_MPI OFF PARENT_SCOPE )
-  if ( with-mpi )
+  if ( NOT "${with-mpi}" STREQUAL "OFF" )
```

If the user has given a directory as option, we need to pass this to CMake so that `find_package` will look in the right place:
```suggestion
  if ( NOT "${with-mpi}" STREQUAL "OFF" )
    if ( NOT ${with-mpi} STREQUAL "ON" )
      # if set, use this prefix
      set( MPI_ROOT_DIR "${with-mpi}" )
    endif ()
```